pkg/providers/nutanix/clientcache.go

```diff
@@ -0,0 +1,65 @@
+package nutanix
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	prismgoclient "github.com/nutanix-cloud-native/prism-go-client"
+	"github.com/nutanix-cloud-native/prism-go-client/environment/credentials"
+	v3 "github.com/nutanix-cloud-native/prism-go-client/v3"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+// ClientCache is a map of NutanixDatacenterConfig name to Nutanix client.
+type ClientCache struct {
+	clients map[string]Client
+}
+
+// NewClientCache returns a new ClientCache.
+func NewClientCache() *ClientCache {
+	return &ClientCache{
+		clients: make(map[string]Client),
+	}
+}
+
+// GetNutanixClient returns a Nutanix client for the given NutanixDatacenterConfig.
+func (cb *ClientCache) GetNutanixClient(datacenterConfig *anywherev1.NutanixDatacenterConfig, creds credentials.BasicAuthCredential) (Client, error) {
+	if client, ok := cb.clients[datacenterConfig.Name]; ok {
+		return client, nil
+	}
+
+	clientOpts := make([]v3.ClientOption, 0)
+	if datacenterConfig.Spec.AdditionalTrustBundle != "" {
+		block, _ := pem.Decode([]byte(datacenterConfig.Spec.AdditionalTrustBundle))
+		certs, err := x509.ParseCertificates(block.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse additional trust bundle %s: %v", datacenterConfig.Spec.AdditionalTrustBundle, err)
+		}
+		if len(certs) == 0 {
+			return nil, fmt.Errorf("unable to extract certs from the addtional trust bundle %s", datacenterConfig.Spec.AdditionalTrustBundle)
+		}
+		clientOpts = append(clientOpts, v3.WithCertificate(certs[0]))
+	}
+
+	endpoint := datacenterConfig.Spec.Endpoint
+	port := datacenterConfig.Spec.Port
+	url := fmt.Sprintf("%s:%d", endpoint, port)
```

Comment:
nit: can you use net.JoinHostPort for this instead? https://pkg.go.dev/net#JoinHostPort